### PR TITLE
Kubelet probes would result in the webhook writing the HTTP status twice

### PR DESCRIPTION
**Pull request by dprotaso**
Doesn't seem like it affected anything - just writes out some extra
log messages

### CODE REVIEW

File: webhook/webhook_test.go [hunk 1]
```diff
@@ -96,3 +98,51 @@ func TestRegistrationStopChanFire(t *testing.T) {
...
+	defer cancel()
+
+	recorder := bombRecorder{ResponseRecorder: httptest.NewRecorder()}
+	probeReq := httptest.NewRequest("GET", "/", nil)
+	probeReq.Header.Set("User-Agent", "kube-probe/1.16")
+
+	webhook.ServeHTTP(&recorder, probeReq)
+
+	if got, want := recorder.Code, http.StatusOK; got != want {
+		t.Fatalf("probe got HTTP status %d - expected %d", got, want)
```
Comment (vagababov) [inline webhook/webhook_test.go:114]:
```suggestion
		t.Fatalf("Probe got HTTP status %d - expected %d", got, want)
```
Same below.

File: webhook/webhook_test.go [hunk 2]
```diff
@@ -96,3 +98,51 @@ func TestRegistrationStopChanFire(t *testing.T) {
...
+	if got, want := recorder.writeCount, 1; got != want {
+		t.Errorf("HTTP status was written %d times - expected only one write", got)
+	}
+
+	// Stop the webhook - which means probes should fail
+	//
+	// The steps below aren't obvious and requires you to
+	// know the implementation details
+	cancel()
+	webhook.Run(ctx.Done())
```
Comment (vagababov) [inline webhook/webhook_test.go:126]: can you just do do `close(webhook.stop)`?

Comment (dprotaso) [reply webhook/webhook_test.go:126]: i'd rather not access the internals directly if I can avoid it

**Pull request merged**